core/src/main/java/com/novoda/noplayer/internal/listeners/AdvertListeners.java

```diff
@@ -22,10 +24,30 @@ public void clear() {
     }
 
     @Override
-    public void onAdvertEvent(String event) {
+    public void onAdvertBreakStart(AdvertBreakId advertBreakId) {
         for (NoPlayer.AdvertListener listener : listeners) {
-            listener.onAdvertEvent(event);
+            listener.onAdvertBreakStart(advertBreakId);
```

Comment:
Did you think about renaming this whole class (I cannot comment there) to `CompositeAdvertListener`, to make it more explicit what it's doing?

Reply:
I don't really follow? Oh because it deals with a group of advert listeners? tbh, this is the project convention, we always do the group as `Listeners` and the single as `Listener`. We could open an issue to update the naming if you think it is confusing though? It is pretty easy to occasionally miss that `s` 😂

Reply:
Yeah, sometimes I forget about this project convention and I find `Composite` easier to understand, as it is a known pattern.
Good to be addressed outside this PR 👍